### PR TITLE
Refactor task services

### PR DESCRIPTION
**Pull request by jrbotros**
Moves `autoSave` functionality into its own Angular service!

### CODE REVIEW

File: orchestra/static/orchestra/task/js/autosave.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            if (!confirm('Your latest changes haven\'t been saved.\n\n' +
+                  'Are you sure you want to leave this page?')) {
+              e.preventDefault();
+            }
+          }
+        });
+      }
+
+      autoSaver.schedule = function() {
+        if (!autoSaver.autoSaveTimer && !autoSaver.is_read_only) {
```
Comment (marcua) [inline orchestra/static/orchestra/task/js/autosave.js]: likely from the scope and not from autosaver?

let's manually test
- autosave works
- save works
- no autosave on readonly tasks

File: orchestra/static/orchestra/task/js/autosave.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
+(function() {
```
Comment (marcua) [inline orchestra/static/orchestra/task/js/autosave.js:1]: nitpick: this is a taskSaver with autosaving functionality.  so perhaps calling it a taskSaver and task_save.js might make more sense.

but that's just one person's opinion :)

File: orchestra/static/orchestra/task/js/autosave.js (hunk 1 above)
Comment (jrbotros) [inline orchestra/static/orchestra/task/js/autosave.js]: :white_check_mark: on both counts!

File: orchestra/static/orchestra/task/js/autosave.js (hunk 2 above)
Comment (jrbotros) [inline orchestra/static/orchestra/task/js/autosave.js:1]: I feel like it's an autoSaver which wraps saving functionality—going to leave it as is, I think :blush:

**Pull request merged**